File: SEP3_UserSystem/src/main/java/model/domain/Unit/User/Account.java
package model.domain.Unit.User;
import model.domain.List.UserList.FriendSettingList;

import java.util.Date;

public class Account extends UserInformation {
    private String Password;
    private FriendSettingList FriendSettingList;

    public Account(String id, String userName,String password) {
        super(id, userName);
        Password = password;
        FriendSettingList = new FriendSettingList();
    }

    private Account(String id, String userName, Date birthday, String password, FriendSettingList friendSettingList) {
        this(id, userName, password);
        setBirthday(birthday);
        FriendSettingList = friendSettingList;
    }

    public String changePassWord(String oldPassword,String newPassword) {
        if (Password.equals(oldPassword))
        {
            Password = newPassword;
            return null;
        }
        else
        {
            return "Wrong password.";
        }
    }

    public FriendSettingList getFriendSettingList() {
        return FriendSettingList;
    }

    public void setFriendSettingList(model.domain.List.UserList.FriendSettingList friendSettingList) {
        FriendSettingList = friendSettingList;
    }

    public void update(Account account)
    {
        update(account.getUserName(),account.getBirthday());
        FriendSettingList = account.getFriendSettingList();
    }

    public Account copy() {
        return new Account(getId(),getUserName(),getBirthday(),Password,FriendSettingList.copy());
    }
}

File: SEP3_PostSystem/Model/Unit/User/Account.cs
using SEP3_PostSystem.Model.List.UserList;

namespace SEP3_PostSystem.Model.Unit.User
{
    public class Account : BasicInformation
    {
        public string Password { get; set; }
        public FriendSettingList FriendSettingList { get; set; }
        public Account(string id, string userName,string password) : base(id, userName)
        {
            Password = password;
            FriendSettingList = new FriendSettingList();
        }

        public Account()
        {
            
        }
        
        private Account(string id, string userName, Birthday birthday, char gender, string homeTown, long phoneNumber, string password, FriendSettingList friendSettingList,UserStatus userStatus) : this(id, userName, password)
        {
            SetBirthday(birthday);
            SetGender(gender);
            SetHomeTown(homeTown);
            SetPhoneNumber(phoneNumber);
            SetUserStatus(userStatus);
            FriendSettingList = friendSettingList;
        }

        public bool CheckPassword(string password)
        {
            return Password.Equals(password);
        }

        public bool CheckPassword(Account otherAccount)
        {
            return otherAccount.GetId().Equals(GetId())&&otherAccount.CheckPassword(Password);
        }

        public string ChangePassWord(string oldPassword,string newPassword) 
        {
            if (Password.Equals(oldPassword))
            {
                Password = newPassword;
                return null;
            }
            else
            {
                return "Wrong password.";
            }
        }
        
        public string ChangePassWord(Account oldAccount,Account newAccount) {
            if (GetId().Equals(newAccount.GetId())&&CheckPassword(oldAccount))
            {
                return ChangePassWord(oldAccount.Password,newAccount.Password);
            }
            else 
            {
                return "Wrong change.";
            }
        }

        public FriendSettingList GetFriendSettingList() 
        {
            return FriendSettingList;
        }

        public void SetFriendSettingList(FriendSettingList friendSettingList) 
        {
            FriendSettingList = friendSettingList;
        }

        public void Update(Account account)
        {
            Update(account.UserName,account.Birthday,account.Gender,account.Hometown,account.PhoneNumber);
            FriendSettingList = account.GetFriendSettingList();
        }
        
        public Account Copy() 
        {
            return new Account(GetId(),GetUserName(),GetBirthday(),Gender,Hometown,PhoneNumber,Password,FriendSettingList.Copy(),UserStatus.Copy());
        }
    }
}